lib/tasks/one_off.rake

```diff
@@ -138,4 +138,35 @@ namespace :pvb do
     FileUtils.rm(password_filename)
     STDOUT.puts "\nRemoved passwords."
   end
+
+  desc 'Backpopulate cancellation reasons'
+  task backpopulate_cancellation_reasons: :environment do
+    missing_cancellation = Visit.
+                           preload(:cancellation).
+                           with_processing_state('cancelled').
+                           joins(<<-EOS).
+        LEFT OUTER JOIN cancellations \
+          ON cancellations.visit_id = visits.id
+      EOS
+                           where(cancellations: { id: nil })
+
+    batch = missing_cancellation.limit(1).to_a
+
+    while batch.any?
+      batch.each do |visit|
+        if visit.cancellation
+          fail "Visit #{visit.id} already has a cancellation!"
+        end
+
+        Cancellation.create!(visit_id: visit.id,
+                             reason: Cancellation::VISITOR_CANCELLED,
+                             created_at: visit.updated_at,
+                             updated_at: visit.updated_at)
+      end
+
+      batch = missing_cancellation.limit(1).to_a
+    end
+
+    STDOUT.puts "Done. Missing cancellation: #{missing_cancellation.count}"
```

Comment:
Mmmh, that will return `0` when it runs successfully. I want to check that there are no visits that have been missed. I'll tweak the message to make it clear.